clients/src/main/java/org/apache/kafka/common/requests/TxnOffsetCommitRequest.java

```diff
@@ -45,9 +45,39 @@
 
         public final TxnOffsetCommitRequestData data;
 
-        public Builder(TxnOffsetCommitRequestData data) {
+        public Builder(final String transactionalId,
```

Comment:
When working with the multi-version API, I realized that by making the data initialization internal could save a lot of caller's effort.